src/kem/kyber768/constant_time_ops.rs

```diff
@@ -0,0 +1,41 @@
+use crate::kem::kyber768::{CIPHERTEXT_SIZE, SHARED_SECRET_SIZE};
+
+// TODO: Attribute this to RustCrypto
+#[inline]
+fn is_non_zero(value: u8) -> u8 {
+    const SHIFT_BITS: usize = core::mem::size_of::<u64>() - 1;
+    let value = value as u64;
+
+    (((value | (!value).wrapping_add(1)) >> SHIFT_BITS) & 1) as u8
+}
+
+pub(crate) fn compare_ciphertexts_in_constant_time(lhs: &[u8], rhs: &[u8]) -> u8 {
```

Comment:
Let's make sure this is not optimized out. From what I see, it doesn't look good. https://godbolt.org/z/obsMY3YYM
I'm fine to leave this in and replace then later with something actually constant time.

Reply:
Added a comment about this.